filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsFileSystemProvider.scala

```diff
@@ -82,12 +83,21 @@ object ExecutionContextExecutorServiceBridge {
   * This implementation is not complete and mostly a proof of concept that it's possible to *copy* around files from/to local/gcs.
   * Copying is the only functionality that has been successfully tested (same and cross filesystems).
   *
+  * If/when switching to Google's GCS NIO implementation, callers may need to implement various utilities built into
+  * this implementation, including:
+  *
+  * - Minimizing the upload buffer size, assuming the default is also on the order of megabytes of memory per upload
+  * - Automatically retrying transient errors
+  * - etc.
+  *
```

Comment:
Yes, I think when we switch to google's NIO we should still write a wrapper around it to take care of these things